pkg/system/phase2_creating.go

```diff
@@ -192,6 +198,19 @@ func (r *Reconciler) SetDesiredServiceS3() error {
 	return nil
 }
 
+// SetDesiredServiceSts updates the ServiceSts as desired for reconciling
+func (r *Reconciler) SetDesiredServiceSts() error {
+	if r.NooBaa.Spec.DisableLoadBalancerService {
+		r.ServiceSts.Spec.Type = corev1.ServiceTypeClusterIP
+	} else {
+		// It is here in case disableLoadBalancerService is removed from the crd or changed to false
+		r.ServiceSts.Spec.Type = corev1.ServiceTypeLoadBalancer
+	}
+	r.ServiceSts.Spec.Selector["noobaa-s3"] = r.Request.Name
+	r.ServiceSts.Labels["noobaa-sts-svc"] = "true"
```

Comment:
the "noobaa-sts-svc" label is already configured in `service-sts.yaml` to true, it will be redundant to set it here too, no?

Reply:
As far as I understand this label is used by the service monitor, probably it's redundant, WDYT?

Reply:
I don't see how this label is used by the service monitor, but either way, setting the label here and also in the yaml file is not required, if the service monitor do use this label then we should set it only once.

Reply:
currently, it's not used by the service monitor, but we might would like to add that in the future
you can take a look at https://github.com/noobaa/noobaa-operator/pull/619 and see that this is due to upgrade
issues we fixed so it's needed

Reply:
I think i'll just remove it in the yaml and in here too